plugins/woocommerce/src/Admin/Features/Settings/Init.php

```diff
@@ -165,18 +165,33 @@ public static function add_component_settings( $settings ) {
 		$setting_pages = \WC_Admin_Settings::get_settings_pages();
 		$pages         = array();
 		foreach ( $setting_pages as $setting_page ) {
-			$scripts_before_adding_settings = $wp_scripts->queue;
-			$pages                          = $setting_page->add_settings_page_data( $pages );
-
+			$scripts_before_adding_settings                         = $wp_scripts->queue;
```

Comment:
This changes the result of `settingsScripts`. The `$scripts_before_adding_settings` is now the same as next line where we compare the difference to know what scripts are needed for the current page.

Reply:
Oh yikes, didn't see that.

Fixed in [keep script diff logic intact](https://github.com/woocommerce/woocommerce/pull/55467/commits/eea96217fb55be67a823f018e19678cc94681b23)